backend/fileSystem.js

```diff
@@ -145,6 +147,84 @@ class FileSystemBackend {
         const match = await this.getPackageByName(packageName);
         return match ? match.versions[version] : null;
     }
+    async searchPackages(properties) {
+        // Retrieve listing of all packages
+        await this.syncMetadata();
+        const packages = Object.keys(this._metadata).map(key => this._metadata[key]);
+
+        // Begin with all packages
+        let result = packages;
+
+        // Filter by search query
+        if (typeof properties.query === 'string' && properties.query.length > 0) {
+            // If only searching for names, it's possible to more efficiently search the whole package array
+            let searchProperties = packages;
+
+            // If doing an advanced search with additional properties, make a new array of package properties
+            if (properties.property.length !== 1 || properties.property[0] !== 'name') {
+                // Extract searchable properties
+                searchProperties = packages.map((pack) => {
+                    // Get latest
+                    const latest = pack.versions[pack['dist-tags'].latest];
+
+                    // Return selected properties
+                    return {
+                        name: pack.name,
+                        description: latest.description || pack.description,
+                        author: latest._npmUser.name,
+                        readme: latest.readme || pack.readme,
+                        keywords: latest.keywords,
+                    };
+                });
+            }
+
+            // Set options for searching
+            const options = {
+                shouldSort: true,
+                threshold: 0.6,
+                maxPatternLength: 32,
+                keys: properties.property,
+            };
+
+            // Perform search for the query in the package array
+            const fuse = new Fuse(searchProperties, options);
+            result = fuse.search(properties.query);
+        }
+
+        // Filter by author
+        if (typeof properties.author === 'string' && properties.author.length > 0) {
+            result = result.filter((pack) => {
+                const latest = pack.versions[pack['dist-tags'].latest];
+                return latest._npmUser === properties.author;
+            });
+        }
+
+        // Sort by recent
+        if (properties.sort === 'recent') {
+            result = result.sort((a, b) => {
+                const aLatestTime = a.time[a['dist-tags'].latest];
+                const bLatestTime = b.time[b['dist-tags'].latest];
+
+                return Date.parse(bLatestTime) - Date.parse(aLatestTime);
+            });
+        }
+
+        // Sanitize the start value
+        let startingValue = properties.start;
+        if (startingValue === undefined || startingValue < 0) {
```

Comment:
Let's condense this into something like:
```
const startingValue = Math.max(properties.start || 0, 0);
```

This should default an empty `start` variable to 0, and also take care of any values that are below 0